Ghidra/Processors/STM/data/languages/ST7.sinc

```diff
@@ -0,0 +1,758 @@
+  I = 0;
+@endif
+@ifdef ST72
+  I0 = 0;
+  I1 = 1;
+@endif
+  halt();
+}
+
+:INC BITOP is kind=0x0C ... & BITOP {
+  local value = BITOP + 1;
+  nzFlags(value);
+  BITOP = value;
+}
+
+:IRET is $(NO_PRE) & opcode=0x80 {
+  SP = SP + 1;
+  CC = *:1 SP;
+  C = CC[0,1];
+  Z = CC[1,1];
+  N = CC[2,1];
+@ifdef ST70
+  I = CC[3,1];
+@endif
+@ifdef ST72
+  I0 = CC[3,1];
+@endif
+  H = CC[4,1];
+@ifdef ST72
+  I1 = CC[5,1];
+@endif
+  SP = SP + 1;
+  A = *:1 SP;
+  SP = SP + 1;
+  X = *:1 SP;
+  SP = SP + 1;
+  PCH = *:1 SP;
+  SP = SP + 1;
+  PCL = *:1 SP;
+}
+
+:JP ADDROP is ADDROP {
+  PC = ADDROP;
+}
+
+:JRA REL is $(NO_PRE) & opcode=0x20 ; REL {
+  PC = PC + REL;
+}
+
+:JRA [imm8] is $(PRE_PIX) & opcode=0x20 ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local offset = sext(*:1 target);
+  PC = PC + offset;
+}
+
+:LD A, ASMEMOP is kind=0x06 ... & ASMEMOP & A {
+  A = ASMEMOP;
+  nzFlags(A);
+}
+
+:LD ADMEMOP, A is kind=0x07 ... & ADMEMOP & A {
+  ADMEMOP = A;
+  nzFlags(A);
+}
+
+:LD X, XSMEMOP is kind=0x0E ... & XSMEMOP & X {
+  X = XSMEMOP;
+  nzFlags(X);
+}
+
+:LD XDMEMOP, X is kind=0x0F ... & XDMEMOP & X {
+  XDMEMOP = X;
+  nzFlags(X);
+}
+
+:LD Y, YSMEMOP is kind=0x0E ... & YSMEMOP & Y {
+  Y = YSMEMOP;
+  nzFlags(Y);
+}
+
+:LD YDMEMOP, Y is kind=0x0F ... & YDMEMOP & Y {
+  YDMEMOP = Y;
+  nzFlags(Y);
+}
+
+:LD X, A is $(NO_PRE) & opcode=0x97 & X & A {
+  X = A;
+  nzFlags(A);
+}
+
+:LD A, X is $(NO_PRE) & opcode=0x9F & X & A {
+  A = X;
+  nzFlags(X);
+}
+
+:LD Y, A is $(PRE_PDY) & opcode=0x97 & Y & A {
+  Y = A;
+  nzFlags(A);
+}
+
+:LD A, Y is $(PRE_PDY) & opcode=0x9F & Y & A {
+  A = Y;
+  nzFlags(Y);
+}
+
+:LD Y, X is $(PRE_PDY) & opcode=0x93 & X & Y {
+  Y = X;
+  nzFlags(X);
+}
+
+:LD X, Y is $(NO_PRE) & opcode=0x93 & X & Y {
+  X = Y;
+  nzFlags(Y);
+}
+
+:LD A, S is $(NO_PRE) & opcode=0x9E & S & A {
+  A = S;
+  nzFlags(S);
+}
+
+:LD S, A is $(NO_PRE) & opcode=0x95 & S & A {
+  S = A;
+  nzFlags(A);
+}
+
+:LD X, S is $(NO_PRE) & opcode=0x96 & X & S {
+  X = S;
+  nzFlags(S);
+}
+
+:LD S, X is $(NO_PRE) & opcode=0x94 & X & S {
+  S = X;
+  nzFlags(X);
+}
+
+:LD Y, S is $(PRE_PDY) & opcode=0x96 & Y & S {
+  Y = S;
+  nzFlags(S);
+}
+
+:LD S, Y is $(PRE_PDY) & opcode=0x94 & Y & S {
+  S = Y;
+  nzFlags(Y);
+}
+
+:MUL X, A is $(NO_PRE) & opcode=0x42 & X & A {
+  H = 0;
+  C = 0;
+  local destination:2 = zext(X);
+  destination = destination * zext(A);
+  local low:2 = destination & 0xFF;
+  local high:2 = (destination >> 8) & 0xFF;
+  A = low:1;
+  X = high:1;
+}
+
+:MUL Y, A is $(PRE_PDY) & opcode=0x42 & Y & A {
+  H = 0;
+  C = 0;
+  local destination:2 = zext(Y);
+  destination = destination * zext(A);
+  local low:2 = destination & 0xFF;
+  local high:2 = (destination >> 8) & 0xFF;
+  A = low:1;
+  Y = high:1;
+}
+
+:NEG BITOP is kind=0x00 ... & BITOP {
+  local value = (BITOP ^ 0xFF) + 1;
+  nzFlags(value);
+  BITOP = value;
+  if !Z goto inst_next;
+  C = 0;
+}
+
+:NOP is $(NO_PRE) & opcode=0x9D {
```

Comment:
i like to put something in here, just so its not literally a nop in ghidra's eyes. (2nd line is to remove compiler warning)
```
local nop:1 = 0;
nop = nop;
```

Reply:
It'd be awesome if the Ghidra team would write a style guide on how to write Sleigh code.  Right now all other files have bare NOPs (the prefix definitions are also treated as NOP, go figure), so I suppose that's just the way it is.

Reply:
I had talked with @emteere  about the this a while ago and we tossed around the idea of a NOP sleigh identifier.  This is really just to remove a warning I think, I can't recall off the top of my head if `{}` has adverse effects in analysis, I think its only `unimpl`